ipl/main.c

```diff
@@ -1969,26 +1985,30 @@ void fix_sd_attr(u32 type)
 	gfx_clear_grey(&gfx_ctxt, 0x1B);
 	gfx_con_setpos(&gfx_con, 0, 0);
 
-	char buff[256];
+	char buff[8];
```

Comment:
Why you changed that? This is the directory buffer that is passed on.
With the first folder/files you gonna have a buffer overflow.

Reply:
You're 3/3 on the bugs. Nice job. I actually wasn't sure what was causing the crashes I was seeing, but your comment makes perfect sense now.